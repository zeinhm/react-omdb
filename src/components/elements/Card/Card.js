import React from 'react';
import PropTypes from 'prop-types';
import './styles.scoped.css';

export default function Card(props) {
  const { children, className } = props;
  const classes = ['card-component', className].filter(Boolean).join(' ');

  return (
    <section className={classes}>
      {children}
    </section>
  );
}

Card.defaultProps = {
  children: null,
  className: '',
};

Card.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};
