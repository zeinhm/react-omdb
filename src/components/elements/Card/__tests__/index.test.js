import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Card from '../Card';

describe('src/components/elements/Card', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Card />);
    expect(tree).toMatchSnapshot();
  });
});
