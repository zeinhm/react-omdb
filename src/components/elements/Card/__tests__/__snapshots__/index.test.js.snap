// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/Card render 1`] = `
<section
  className="card-component"
/>
`;
